mod_reforged/hooks/skills/items/firearms_resistance_skill.nut

```diff
@@ -0,0 +1,14 @@
+::mods_hookExactClass("skills/items/firearms_resistance_skill", function(o) {
+    // We overwrite the vanilla behavior of that skill completely while replicating its effect pretty future proof
+    o.onBeforeDamageReceived = function( _attacker, _skill, _hitInfo, _properties )
+	{
+        if (_skill.getItem().isWeaponType(::Const.Items.WeaponType.Firearm))    // This covers all kinds of firearms
+        {
+            _properties.DamageReceivedRegularMult *= 0.66;
+        }
+        else if (_hitInfo.DamageType == ::Const.Damage.DamageType.Burning)  // This covers burning ground and firelances
```

Comment:
Just want to point out that in Reforged I've changed the `WeaponType` of Firelance to be `Spear` only. I've removed the `Firearm` type from it because it conflicts with many things e.g. weapon masteries and this system. The `firearm` skill on that weapon is just a one-off use, so I think Spear fits it better anyway.

Reply:
Ok, I never expected it to be firearm in the first place.
And I already account for burning damage which the ignite firelance skill applies exclusively.

So I don't see anything that needs to be resolved here

Reply:
Yeah, I just wanted to mention that in case you didn't know it. Nothing to Resolve.